RAGchain/utils/linker/dynamo_linker.py

```diff
@@ -102,15 +103,29 @@ def handle_error(err, table_name, operation):
 
     def get_json(self, ids: list[Union[UUID, str]]):
         str_ids = [str(find_id) for find_id in ids]
-        response_list = [self.table.get_item(Key={'id': find_id}) for find_id in str_ids]
-        return [response['Item']['db_origin'] for response in response_list]
+        data_list = []
+        for find_id in str_ids:
+            # Check if id exists in dynamo linker
+            """
+            response_list = [self.table.get_item(Key={'id': find_id}) for find_id in str_ids]
+            return [response['Item']['db_origin'] for response in response_list]
+            """
+            data = self.table.get_item(Key={'id': find_id})
+            if 'Item' not in data:
+                warnings.warn(f"ID {find_id} not found in DynamoLinker", NoIdWarning)
+            else:
+                db_origin = data['Item']['db_origin']
+                # Check if data exists in dynamo linker
+                if db_origin is None:
+                    warnings.warn(f"Data {find_id} not found in DynamoLinker", NoDataWarning)
+                    continue
+                else:
+                    data_list.append(db_origin)
+        return data_list
 
     def flush_db(self):
         self.table.delete()
 
-    def __del__(self):
-        self.dynamodb.close()
```

Comment:
Why do you delete this code? It looks good..

Reply:
dynamodb doesn't support .close(). It's copilot's halucination :(

Reply:
Then I think there is some way to close dynamodb connection?